core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -684,17 +689,16 @@ class KafkaServer(val config: KafkaConfig, time: Time = Time.SYSTEM, threadNameP
   }
 
   private def checkpointBrokerId(brokerId: Int) {
```

Comment:
I think the purpose of KIP-112 is to make sure that broker be able to serve replicas on the good log directory even if there is bad log directory. We can not achieve this goal and the availability will be reduced if broker can not startup due to disk failure.

Can you explain a bit more why it is a concern if broker id load fails in one log directory? Does this cause data corruption?